src/concrete_convex_hull.jl

```diff
@@ -186,6 +191,126 @@ function _three_points_2d!(points::AbstractVector{<:AbstractVector{N}}) where {N
     return points
 end
 
+function _four_points_2d!(points::AbstractVector{<:AbstractVector{N}}) where {N<:Real}
+    A, B, C, D = points[1], points[2], points[3], points[4]
+    tri_ABC = (A[2] - B[2]) * C[1] + (B[1] - A[1]) * C[2] + (A[1] * B[2] - B[1] * A[2])
```

Comment:
Define a helper function and use the factored form:

```julia
@inline _orientation(A, B, C) = (A[2] - B[2]) * (C[1] - A[1]) + (B[1] - A[1]) * (C[2] - A[2])
```

Reply:
.. or (better) use [right_turn](https://github.com/JuliaReach/LazySets.jl/blob/master/src/concrete_convex_hull.jl#L243) 😄

Reply:
Hm, the three patterns do not match. They might still be equivalent, but it is not obvious to me.

Reply:
Let:
```julia
    tri_ABC = (A[2] - B[2]) * C[1] + (B[1] - A[1]) * C[2] + (A[1] * B[2] - B[1] * A[2]) 
... adding and substracting:  
= (A[2] - B[2]) * C[1] + (B[1] - A[1]) * C[2] + A[1] * B[2] - B[1] * A[2] + A[2]*A[1] - A[2]*A[1]
= (A[2] - B[2]) * C[1] + (B[1] - A[1]) * C[2] - (A[2] - B[2])(A[1]) - (B[1] - A[1])*(A[2])
= (A[2] - B[2])*(C[1] - A[1]) + (B[1] - A[1]) * (C[2] - A[2])
```
while:
```
right_turn(O, A, B) = (A[1] - O[1]) * (B[2] - O[2]) - (A[2] - O[2]) * (B[1] - O[1])
=>
right_turn(A, B, C) = (B[1] - A[1]) * (C[2] - A[2]) - (B[2] - A[2]) * (C[1] - A[1])
                               =  (A[2] - B[2]) * (C[1] - A[1]) + (B[1] - A[1]) * (C[2] - A[2])  
```
=> `tri_ABC = right_turn(A, B, C)`.